### PR TITLE
Bug/3237 creating legend with loc=best is possibly slowing down track visualization

### PR DESCRIPTION
**Pull request by randy-seng**
OP#3237

### CODE REVIEW

File: OTAnalytics/plugin_prototypes/track_visualization/track_viz.py [hunk 1]
```diff
@@ -519,8 +518,16 @@ def _plot_dataframe(self, track_df: DataFrame, axes: Axes) -> None:
             alpha=self._alpha,
             ax=axes,
             palette=self._color_palette_provider.get(),
-            legend=self._enable_legend,
         )
+        if self._enable_legend:
+            print("creating legend")
```
Comment (martinbaerwolff) [inline OTAnalytics/plugin_prototypes/track_visualization/track_viz.py:523]: Can this print be removed?

File: OTAnalytics/plugin_prototypes/track_visualization/track_viz.py [hunk 2]
```diff
@@ -519,8 +518,16 @@ def _plot_dataframe(self, track_df: DataFrame, axes: Axes) -> None:
             alpha=self._alpha,
             ax=axes,
             palette=self._color_palette_provider.get(),
-            legend=self._enable_legend,
         )
+        if self._enable_legend:
```
Comment (martinbaerwolff) [inline OTAnalytics/plugin_prototypes/track_visualization/track_viz.py:522]: Where did self._enable_legend come from? Is this for a future feature to toggle the legend?

File: OTAnalytics/plugin_prototypes/track_visualization/track_viz.py [hunk 3]
```diff
@@ -519,8 +518,16 @@ def _plot_dataframe(self, track_df: DataFrame, axes: Axes) -> None:
...
+            print("creating legend")
+            legend = axes.legend(loc="upper right")
+            legend.set_alpha(1)
+        else:
+            # Somehow enabling a legend from previous call to this method persists.
```
Comment (martinbaerwolff) [inline OTAnalytics/plugin_prototypes/track_visualization/track_viz.py:527]: Is this (maybe) because the axes persist, too?

**Review by martinbaerwolff — CHANGES_REQUESTED** [review]
Looks good, just one small thing and two questions just for information.

File: OTAnalytics/plugin_prototypes/track_visualization/track_viz.py (hunk 3 above)
Comment (randy-seng) [reply OTAnalytics/plugin_prototypes/track_visualization/track_viz.py:527]: I just had a look at the code. The `track_viz.MatplotlibTrackPlotter.plot`  method creates the axes via `_create_figure` and `_create_axes` methods. Nothing indicates that existing axes are being used, at least to me 🤔

File: OTAnalytics/plugin_prototypes/track_visualization/track_viz.py (hunk 2 above)
Comment (randy-seng) [reply OTAnalytics/plugin_prototypes/track_visualization/track_viz.py:522]: `_enable_legend` is a member variable of `TrackGeometryPlotter`. In `main_application.py` several instances of `TrackGeometryPlotter` are created and passed to several layers that want to plot tracks on the canvas. Thus, we don't differentiate between plotters used in the different layers using it. Hence, the need to for `_enable_legend`. If you have a look at the `_create_layers` method in `main_application.py` you can see how plotters are being configured.

In the future, the member variable will be not necessary anymore, if the legend is displayed outside of the canvas.

File: OTAnalytics/plugin_prototypes/track_visualization/track_viz.py (hunk 3 above)
Comment (martinbaerwolff) [reply OTAnalytics/plugin_prototypes/track_visualization/track_viz.py:526]: Hm ok, anyway, we have this in mind in case of plotting issues :)

**Review by martinbaerwolff — APPROVED** [review]

**Pull request merged**